src/ndarray/ndarray.cc

```diff
@@ -1592,8 +1601,12 @@ void NDArray::Save(dmlc::Stream *strm) const {
     save_data = nd_cpu.data();
   } else {
     this->WaitToRead();
-    save_data = this->data();
     nd_cpu = *this;
+#if MXNET_USE_MKLDNN == 1
+    if (nd_cpu.IsMKLDNNData())
```

Comment:
NIt: I believe the project encourages braces even on single line conditionals.

Reply:
I don't see such rule from the code itself. Single line conditionals without braces can be found everywhere, even in this file. See line 1591, 1654, 1669 ...